target/scripts/start-mailserver.sh

```diff
@@ -170,24 +169,35 @@ function _register_functions() {
 # ? >> Executing all stacks / actual start of DMS
 # ------------------------------------------------------------
 
-_early_supervisor_setup
-_early_variables_setup
+# fresh container?
+if [[ ! -f /CONTAINER_START ]]; then
+  # first container start
+  _early_supervisor_setup
+  _early_variables_setup
 
-_log 'info' "Welcome to docker-mailserver ${DMS_RELEASE}"
+  _log 'info' "Welcome to docker-mailserver ${DMS_RELEASE}"
 
-_register_functions
-_check
-_setup
-[[ ${LOG_LEVEL} =~ (debug|trace) ]] && print-environment
-_run_user_patches
-_start_daemons
+  _register_functions
+  _check
+  _setup
+  _run_user_patches
+else
+  # container was restarted
+  _early_variables_setup
+
+  _log 'info' 'Container was restarted. Skipping setup routines..'
+  _log 'info' "Welcome to docker-mailserver ${DMS_RELEASE}"
+
+  _register_functions
```

Comment:
It seems like the two `_early` methods could be handled before this conditional branch? `_early_supervisor_setup` isn't doing anything of concern with a container restart:

https://github.com/docker-mailserver/docker-mailserver/blob/267fc552d220414be9f8ba7dbb2f0ed2431f3228/target/scripts/startup/setup-stack.sh#L37-L62

While the only reason you're calling `_early_variables_setup` + `_register_functions` here is for the later `_start_daemons`? (_the later `print-environment` call is probably less meaningful with restarts, it may even be more helpful for us knowing that it's only output when the user ensures a fresh container?_)

---

So realistically, if the container is fresh we want to run:

```bash
_check
_setup
_run_user_patches

[[ ${LOG_LEVEL} =~ (debug|trace) ]] && print-environment
```

Otherwise:

```bash
_log 'info' 'Container was restarted. Skipping setup routines..'
```

Do you think that makes more sense to reduce the conditional scope to just that?